File: lib/nls_signal.c
/*
% File:	    nls_signal.c
% Authors:  Lan
% Purpose:  Provides signal handling
*/

#include <signal.h>
#include "sicstus/sicstus.h"
#include "nls_signal_glue.h" 

/* External interface */

/* long C_establish_signal_handling(void); */
void abort_handler(int arg);


/*****************************************************************************/
/* Establish signal handlers.
*/

long C_establish_signal_handling(void)
{

	signal(SIGTERM, abort_handler);
	signal(SIGPIPE, abort_handler);
	return 1;

} /* C_establish_signal_handling */

void abort_handler(int arg)
{
	/* SP_raise_fault("ABORT"); */
	abort();

} /* abort_handler */

File: lib/nls_signal.pl
% File:	    nls_signal.pl
% Module:   NLS Signal
% Author:   Lan
% Purpose:  Provide signal handling


:- module(nls_signal,[
    establish_signal_handling/0
    ]).


foreign_resource(nls_signal, ['C_establish_signal_handling']).

foreign('C_establish_signal_handling', c,
        'C_establish_signal_handling'([-integer])).

:- load_foreign_resource(nls_signal).

% :- abolish(foreign_resource/2, [force(true)]).

% :- abolish(foreign/3, [force(true)]).

/* establish_signal_handling

establish_signal_handling/0 calls C_establish_signal_handling which calls
signal() to establish signal handlers.  */

establish_signal_handling :-
    'C_establish_signal_handling'(1).
